### PR TITLE
IP-348 - TieredVariantInjectRD Generation

### PR DESCRIPTION
**Pull request by greglever**
[IP-348 - TieredVariantInjectRD Generation](https://jira.extge.co.uk/browse/IP-348)
=============================================================
We need to validate `TieredVariantInjectRD` objects in CVA and the CIP API and the validation service needs to generate these objects for its tests

Summary of Changes
================
* Add `get_valid_tiered_variant_inject_rd_0_3_1` and `get_valid_tiered_variant_inject_rd_0_4_0`
* Add `test_tiered_variant_inject_rd` (for `cva_0_3_1` and `cva_0_4_0_SNAPSHOT`) to `test_generate_mock_objects.py`

Tests passing locally
================
```
(.env) $ python -m unittest discover
.................................................
----------------------------------------------------------------------
Ran 49 tests in 2.334s

OK
(.env) $ 
```

### CODE REVIEW

**Comment by cboustred** [comment]
These seem like some quite hefty updates which I'm not 100%
 confident reviewing e.g. [migrating interpreted genomes and report events](https://github.com/genomicsengland/GelReportModels/pull/134/files#diff-b506bf55d7f770a9f7137c2f7619b0ffR54). @greglever do you have some time to explain to me a bit more whats happening?

**Comment by greglever** [comment]
yup @cboustred those are the commits included from https://github.com/genomicsengland/GelReportModels/pull/130 so it may be easier to review that one first. Ideally if we start reviewing the oldest PR first then it makes more sense :smile:

File: protocols/util/generate_mock_objects.py [hunk 1]
```diff
@@ -1345,3 +1347,38 @@ def get_valid_cancer_interpretation_request_2_1_0():
     new_cir.reportVersion = 1
 
     return validate_object(object_to_validate=new_cir, object_type=object_type)
+
+
+def get_valid_tiered_variant_inject_rd_0_3_1():
```
[the remark this answers is not in the document]
Comment (priesgo) [reply protocols/util/generate_mock_objects.py]: here @greglever !!

File: protocols/util/generate_mock_objects.py [hunk 2]
```diff
@@ -1345,3 +1347,38 @@ def get_valid_cancer_interpretation_request_2_1_0():
...
+    new_tvi_rd.cohortId = ''
+    new_tvi_rd.author = ''
+    new_tvi_rd.authorVersion = ''
+    new_tvi_rd.workspace = ['']
+    new_tvi_rd.interpretationRequest = get_valid_interpretation_request_rd_3_1_0()
+
+    return validate_object(object_to_validate=new_tvi_rd, object_type=object_type)
+
+
+def get_valid_tiered_variant_inject_rd_0_4_0():
```
[the remark this answers is not in the document]
Comment (priesgo) [reply protocols/util/generate_mock_objects.py]: and here @greglever !

**Pull request merged**